### PR TITLE
[ENHANCEMENT] Add expect_compound_columns_to_be_unique expectation

### PR DESCRIPTION
**Pull request by anthonyburdi**
Changes proposed in this pull request:
- Add a new expectation with Pandas backend `expect_compound_columns_to_be_unique` to test uniqueness across several columns (e.g. to check a multi-column primary key)
- Tests and docs for this expectation
- DEPRECATION: Rename expectation `expect_multicolumn_values_to_be_unique` to `expect_select_column_values_to_be_unique_within_record`. Keep `expect_multicolumn_values_to_be_unique` around but have it just print a deprecation warning and run the new expectation.

Addresses issue [# 1807](https://github.com/great-expectations/great_expectations/issues/1807) and issue [#236](https://github.com/great-expectations/great_expectations/issues/236)

Previous Design Review notes:
- TBD
-
-

Thank you for submitting!

### CODE REVIEW

File: great_expectations/core/__init__.py [hunk 1]
```diff
@@ -1076,7 +1076,19 @@ class ExpectationConfiguration(DictDot):
...
+            "default_kwarg_values": {
+                "row_condition": None,
+                "condition_parser": "pandas",
+                "ignore_row_if": "all_values_are_missing",
+                "result_format": "BASIC",
+                "include_config": True,
+                "catch_exceptions": False,
+            },
+        },
+        "expect_compound_columns_to_be_unique": {
```
Comment (alexsherstinsky) [inline great_expectations/core/__init__.py:1098]: Does "mostly" make sense for this expectation?..

File: tests/test_definitions/multicolumn_map_expectations/expect_compound_columns_to_be_unique.json [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+              "b": 1.0
+            }
+          ],
+          "unexpected_index_list": [0,1,2,3,4,5,6,7,8]
+        }
+      },
+    "suppress_test_for": ["sqlalchemy", "sqlite", "postgresql", "mysql", "mssql", "spark"]
+    }]
+  }]
+}
```
Comment (alexsherstinsky) [inline tests/test_definitions/multicolumn_map_expectations/expect_compound_columns_to_be_unique.json:197]: Curious: How was this JSON test file generated?  Automatically or manually?  Also, how come we are excluding databases.  Finally, if we are excluding databases, maybe just suppressing "sqlalchemy" would be sufficient?

**Review by alexsherstinsky — COMMENTED** [review]
Very interesting stuff!  Thanks for inviting me to review!

File: great_expectations/core/__init__.py (hunk 1 above)
Comment (anthonyburdi) [reply great_expectations/core/__init__.py:1098]: Added! Thanks for the suggestion :)

**Comment by jcampbell** [comment]
@anthonyburdi -> I'm excited for this one, but I've been hung up on the breaking-ness. Couldn't we just pass through to the old name and rewrite the name (similarly to how we handle the multi-typed expect_column_values_to_be_of_type in Pandas) and print a deprecation warning?

**Comment by anthonyburdi** [comment]
> @anthonyburdi -> I'm excited for this one, but I've been hung up on the breaking-ness. Couldn't we just pass through to the old name and rewrite the name (similarly to how we handle the multi-typed expect_column_values_to_be_of_type in Pandas) and print a deprecation warning?

Makes sense! The concern initially was that some users were confused by the existing name, so perhaps we can mitigate by making the deprecation note describe the change.

**Comment by anthonyburdi** [comment]
@jcampbell - were you thinking of something like this commit? https://github.com/great-expectations/great_expectations/pull/1822/commits/bf03bf37e2d86af707d157fd2b57be43237b8588 
And where there is the TODO to actually rename all expectations for the user to the new style (e.g. like expect_column_values_to_be_of_type in Pandas)?

**Comment by jjin-ut** [comment]
Oh wow this is great! I'm looking forward to this new feature! We are implementing GE and have a use-case for this right now! Thank you all so much for the great work!!

**Review by abegong — APPROVED** [review]
This looks great. No changes that I can see!

**Pull request merged**